pennylane/transforms/decompositions/clifford_plus_t/cliffordt_transform.py

```diff
@@ -0,0 +1,368 @@
+    ]
+    pauli_hams = [
+        qml.pauli.pauli_sentence(op).hamiltonian(wire_order=range(num_qubits)) for op in pauli_qubit
+    ]
+
+    # Perform U@P@U^\dagger and check if the result exists in set P
+    pauli_coves = []
+    for prod in product([pauli_terms], pauli_hams, [pauli_terms]):
+        # hopefully op_math.prod scales better than matrix multiplication, i.e., O((2^N)^3)
+        upu = qml.pauli.pauli_sentence(qml.prod(*prod))
+        upu.simplify()
+
+        upu2 = upu.hamiltonian(wire_order=range(num_qubits))
+        if not len(upu2.ops) == 1:  # Pauli sum always lie outisde set P
+            return False  # early stopping
+        if not isinstance(upu2.ops[0], qml.Identity):  # Identity always lie in set P
+            pauli_coves.append(any((qml.equal(upu2.ops[0], tm) for tm in pauli_hams)))
+
+    return all(pauli_coves)
+
+
+def check_clifford_t(op):
+    r"""Checks whether the gate is in the standard Clifford+T basis.
+
+    For a given unitary operator :math:`U` acting on :math:`N` qubits, which is not a T-gate,
+    this method checks that the transformation :math:`UPU^{\dagger}` maps the Pauli tensor products
+    :math:`P = {I, X, Y, Z}^{\otimes N}` to Pauli tensor products with :math:`O(N \times 8^N)` time
+    complexity when using naive matrix multiplication.
+
+    Args:
+        op: the operator that needs to be checked
+
+    Returns:
+        Bool that represents whether the provided operator is Clifford+T or not.
+    """
+
+    # Save time and check from the pre-computed list
+    if any(
+        (
+            isinstance(op, gate) or isinstance(getattr(op, "base", None), gate)
+            for gate in _CLIFFORD_T_GATES
+        )
+    ):
+        return True
+    return check_clifford_op(op)
+
+
+def _simplify_param(theta, gate):
+    r"""Check if the parameter allows simplification for the rotation gate.
+
+    For the cases where theta is an integer multiple of π: (a) returns a global phase
+    when even, and (b) reutrns combination of provided gate with global phase when odd.
+    In rest of the other cases it returns None.
+    """
+    if qml.math.isclose(theta, 0.0, atol=1e-6):
+        return [qml.GlobalPhase(0.0)]
+
+    rem, mod = theta / math.pi, theta % math.pi
+    if qml.math.isclose(mod, 0.0, atol=1e-6):
+        ops = [qml.GlobalPhase(theta / 2)]
+        if qml.math.isclose(rem % 2, 1.0, atol=1e-6):
+            ops.append(gate)
+        return ops
+
+    return None
+
+
+def _rot_decompose(op):
+    """Decompose a rotation operation using combination of RZ, S and Hadamard"""
+    d_ops = []
+    # Extend for Rot operation with RzRyRz decompositions
+    if isinstance(op, qml.Rot):
+        for dec in op.decomposition():
+            d_ops.extend(_rot_decompose(dec))
+
+        return d_ops
+
+    (theta,), wires = op.data, op.wires
+    if isinstance(op, qml.RX):
+        ops_ = _simplify_param(theta, qml.PauliX(wires=wires))
```

Comment:
looks like we could pull this line and the `is None` check outside the `if-elif-elif` block?

Reply:
I had thought of doing that using a couple of dictionaries, but I find the code to be more readable this way.